crates/backend/src/codegen.rs

```diff
@@ -899,6 +899,7 @@ impl TryToTokens for ast::ImportFunction {
             abi_arguments.push(quote! { #exn_data_ptr: *mut u32 });
             convert_ret = quote! { Ok(#convert_ret) };
             exceptional_ret = quote! {
+                #[allow(clippy::*)]
```

Comment:
Ah I see -- we should only add `#[allow(clippy::*)]` to things that are top-level items in the expanded code, not anything that is at the top of a `quote!` macro invocation, since it might take multiple quotes to build up a top-level item that we are emitting.

A good rule of thumb is that we should add it on each `fn` we define.

Does that make sense?

Reply:
Yes!!  I understand you said.
I tried to add `#[allow(clippy::*)] ` to things that are top-level items _at first_,
but this code occurred another error like this.

```
error: expected one of `(`, `)`, `,`, or `=`, found `::`
  --> crates/backend/src/codegen.rs:15:15
   |
15 | #[allow(clippy::*)]
   |               ^^ expected one of `(`, `)`, `,`, or `=` here

error: unexpected token: `)`
  --> crates/backend/src/codegen.rs:15:18
   |
15 | #[allow(clippy::*)]
   |                  ^ unexpected token after this

error: aborting due to 2 previous errors
```
　
I found  that  I should use `#[allow(clippy::drop_ref)]` instead of `#[allow(clippy::*)]` when I read codes written by others 🧐💡
Finally, I fixed it.

Reply:
@T5uku5hi, did you try `#![allow(clippy::all)]`?

Reply:
@limira 
Oh! I didn't try `all` instead of `*`.
I tried it just now and I well did it!

I'll add `#[allow(clippy::all)]` in all the top-level items.
Thank you for your kindness 😊

Reply:
This is my bad, sorry!

For some reason I thought the syntax was `allow(clippy::*)` and it is actually `allow(clippy::all)` as @limira points out. Sorry about that!